### PR TITLE
Fix warnings found by flake8

### PR DESCRIPTION
**Pull request by jfbastien**
flake8 checks Python's pep8 style guide and finds bugs along the way.

See: https://flake8.readthedocs.io/en/latest/

It was complaining about the following which I've fixed:

      1  E125 continuation line with same indent as next logical line
      1  E221 multiple spaces before operator
      1  E712 comparison to True should be 'if cond is True:' or 'if cond:'
      1  F401 'collections' imported but unused
      1  F401 'deque' imported but unused
      1  F401 'enum' imported but unused
      1  F401 'hashlib' imported but unused
      1  F401 'importlib' imported but unused
      1  F401 'lxml' imported but unused
      1  F401 're' imported but unused
      1  F401 'SortedList' imported but unused
      1  F401 'sys' imported but unused
      1  F401 'treewalkers' imported but unused
      1  F811 redefinition of unused 'collections' from line 6
      1  F812 list comprehension redefines 'n' from line 363
      1  F821 undefined name 'die'
      1  F841 local variable 'aliases' is assigned to but never used
      1  F841 local variable 'classList' is assigned to but never used
      1  F841 local variable 'failures' is assigned to but never used
      1  F841 local variable 'includeHashes' is assigned to but never used
      1  F841 local variable 'inComment' is assigned to but never used
      1  F841 local variable 'last' is assigned to but never used
      1  F841 local variable 'numfiles' is assigned to but never used
      1  F841 local variable 'obj_method' is assigned to but never used
      1  F841 local variable 'requiredFields' is assigned to but never used
      1  W602 deprecated form of raising exception
      2  E712 comparison to False should be 'if cond is False:' or 'if not cond:'
      2  F401 'closing' imported but unused
      2  F401 'copy' imported but unused
      2  F401 'date' imported but unused
      2  F401 'json' imported but unused
      2  F401 'urllib2' imported but unused
      2  F841 local variable 'outputFile' is assigned to but never used

There are still quite a few left, mostly about whitespace (who says it
isn't significant?). I've silenced the whitespace issues in setup.cfg,
and left the following unfixed for now:

      1  F812 list comprehension redefines 'url' from line 606
      1  F403 'from pygments.lexer import *' used; unable to detect undefined names
      1  F403 'from pygments.token import *' used; unable to detect undefined names
      1  F403 'from repository import *' used; unable to detect undefined names
      3  F403 'from itertools import *' used; unable to detect undefined names
      9  F403 'from htmlhelpers import *' used; unable to detect undefined names
     19  F403 'from messages import *' used; unable to detect undefined names

I also added a Travis CI check using flake8.

### CODE REVIEW

**Comment by jfbastien** [comment]
My editor was complaining about too much stuff!

Let me know if you'd like me to split this up, or if any should be fixed differently. It may be good to fix the remaining F812 but it's ugly because python scoping. We could just silence F403?

File: bikeshed/HTMLSerializer.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # -*- coding: utf-8 -*-
 
 from __future__ import division, unicode_literals
-import re
 import StringIO
 from .htmlhelpers import childNodes, isElement, outerHTML, escapeHTML, escapeAttr
+from .messages import die
```
Comment (tabatkins) [inline bikeshed/HTMLSerializer.py]: Everywhere I use messages, I import the whole set with `import *`. No sense being different here.

Comment (jfbastien) [inline bikeshed/HTMLSerializer.py]: Done.

**Comment by jfbastien** [comment]
Yup, ` ; true` hacks around flake8's return for now. I'll figure out what's wrong later, but at least it's running!

**Pull request merged**